examples/tv_minimization_reconstruction_pdhg.py

```diff
@@ -0,0 +1,190 @@
+
+# %%
+# Set up direct reconstruction class to estimate coil sensitivity maps, density compensation etc
+direct_reconstruction = mrpro.algorithms.reconstruction.DirectReconstruction(kdata)
+img_direct = direct_reconstruction(kdata).data
+
+# run tierative SENSE to get an initiual guess of the solution
+iterative_sense_reconstruction = mrpro.algorithms.reconstruction.IterativeSENSEReconstruction(
+    kdata, n_iterations=8, csm=direct_reconstruction.csm
+)
+img_iterative_sense = iterative_sense_reconstruction(kdata).data
+
+# Define Fourier operator using the trajectory and header information in kdata
+fourier_operator = mrpro.operators.FourierOp.from_kdata(kdata)
+
+# estimate density compensation
+dcf_operator = mrpro.data.DcfData.from_traj_voronoi(kdata.traj).as_operator()
+
+# Calculate coil maps
+img_coilwise = mrpro.data.IData.from_tensor_and_kheader(*fourier_operator.H(*dcf_operator(kdata.data)), kdata.header)
+csm_operator = mrpro.data.CsmData.from_idata_walsh(img_coilwise).as_operator()
+
+# Create the entire acquisition operator A
+acquisition_operator = fourier_operator @ csm_operator
+
+# adjoint and iterative SENSE reconstruction as comparison methods
+# adjoint reconstruction
+(img_adjoint,) = acquisition_operator.H(kdata.data)
+
+
+# %% [markdown]
+# ### Recast the problem to be able to apply PDHG
+
+# %%
+# Define the gradient operator
+from mrpro.operators import FiniteDifferenceOp
+
+# The operator computes the directional derivatives along the penultimate and last dimensions (x,y)
+nabla_operator = FiniteDifferenceOp(dim=(-2, -1), mode='forward')
+
+# Set up the problem by using the previously described identification
+from mrpro.algorithms.optimizers import pdhg
+from mrpro.operators import LinearOperatorMatrix, ProximableFunctionalSeparableSum
+from mrpro.operators.functionals import L1NormViewAsReal, L2NormSquared, ZeroFunctional
+
+# Regularization parameter for the $\ell_1$-norm
+regularization_parameter = 0.5
+
+# Define the functionals
+kdata_tensor = kdata.data
+l2 = 0.5 * L2NormSquared(target=kdata_tensor, divide_by_n=True)
+l1 = regularization_parameter * L1NormViewAsReal(divide_by_n=True)
+
+# Define the functionals f and g and the operator K
+f = ProximableFunctionalSeparableSum(l2, l1)
+g = ZeroFunctional()
+operator = LinearOperatorMatrix(((acquisition_operator,), (nabla_operator,)))
+
+# initialize PDHG with iterative SENSE solution for warm start
+initial_values = (img_iterative_sense.data,)
+
+# %% [markdown]
+# ### Run PDHG for a certain number of iterations
+
+# %%
+max_iterations = 64
+(img_pdhg,) = pdhg(f=f, g=g, operator=operator, initial_values=initial_values, max_iterations=max_iterations)
+
+# %%
+# ### Compare the results
+# Display the reconstructed images
+import matplotlib.pyplot as plt
+
+fig, ax = plt.subplots(1, 3, squeeze=False)
+clim = [0, 1e-3]
+ax[0, 0].set_title('Adjoint Recon', fontsize=10)
+ax[0, 0].imshow(img_direct.abs()[0, 0, 0, :, :], clim=clim)
```

Comment:
if you dont do .data before, you can also just use img_direct.rss() to obtain something plottable

Reply:
but then they differ, no? as far as I understand it, direct reconstruction is x_direct = A^H y (i.e. it depends on the csms), while img_direct.rss() should be (at least from the name) the RSS-recon?